src/main/java/org/jenkinsci/plugins/githubissues/GitHubIssueNotifier.java

```diff
@@ -181,15 +182,15 @@ public void perform(
                 logger.format("GitHub Issue Notifier: Build has started failing, filed GitHub issue #%s%n", issue.getNumber());
                 run.addAction(new GitHubIssueAction(issue, GitHubIssueAction.TransitionAction.OPEN));
             }
-        } else if (result == Result.SUCCESS && issue != null && issue.getState() == GHIssueState.OPEN) {
+        } else if ((result == Result.SUCCESS || result == null) && issue != null && issue.getState() == GHIssueState.OPEN) {
```

Comment:
In declarative pipelines, the result will be `null` unless explicitly set. The common pattern is to set the failure state within a `catch` block, so we can treat `null` as `SUCCESS`. I've added an explanatory comment.